packages/flutter/lib/src/material/ink_decoration.dart

```diff
@@ -252,16 +252,21 @@ class _InkState extends State<Ink> {
   }
 
   Widget _build(BuildContext context, BoxConstraints constraints) {
+    // if _ink have IndexedStack ancestor(s) and not on the display path,
+    // _ink should not paint.
+    final bool inkNeedsPaint = IndexedStack.addIndexedStackScopeAncestorsDependencies(context);
     if (_ink == null) {
       _ink = InkDecoration(
         decoration: widget.decoration,
         configuration: createLocalImageConfiguration(context),
         controller: Material.of(context),
         referenceBox: context.findRenderObject() as RenderBox,
         onRemoved: _handleRemoved,
+        needPaint: inkNeedsPaint,
```

Comment:
This api feels weird, you may as well don't create an ink if inkNeedsPaint = false

Reply:
OK